CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+
+* Submit trivial fixes (for example, a typo, a broken link or an obvious error).
+* Start work on a contribution that was already asked for, or that you’ve already discussed, in an issue.
+
+A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
+
+Always reference the issue related to the PR with `#number-of-issue` and use the markdown to make your comments more readable.
+
+> **Note:** If this is your first pull request, check out [Make a Pull Request](http://makeapullrequest.com/), which [**@kentcdodds**](https://github.com/kentcdodds) created as a walkthrough video tutorial. You can also practice making a pull request in the [First Contributions](https://github.com/Roshanjossey/first-contributions) repository, created by [**@Roshanjossey**](https://github.com/Roshanjossey).
+
+## Coding conventions
+
+**Adhere to standard Python style guidelines**
+All contributions to Netpyne are checked against style guidelines described in PEP 8. We also check for common coding errors (such as variables that are defined but never used).
+
+Several text editors or IDEs also have Python style checking, which can highlight style errors while you code (and train you to make those errors less frequently). This functionality is built-in to the Spyder IDE, but most editors have plug-ins that provide similar functionality. Search for `python linter <name of your favorite editor>` to learn more.
+
+**Use consistent variable naming**
+Classes should be named using `CamelCase`. Functions and instances/variables should use `snake_case` (`n_samples` rather than `nsamples`). Avoid single-character variable names, unless inside a [comprehension](https://docs.python.org/3/glossary.html#term-list-comprehension) or [generator](https://docs.python.org/3/tutorial/classes.html#tut-generators).
+
+**Follow NumPy style for docstrings**
+In most cases imitating existing docstrings will be sufficient, but consult the [Numpy docstring style guidelines](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) for more complicated formatting such as embedding example code, citing references, or including rendered mathematics. Private function/method docstrings may be brief for simple functions/methods, but complete docstrings are appropriate when private functions/methods are relatively complex.
+
+**Other style guidance**
+
+* Add description, arguments and returns (I/O params) to all functions and methos.
+* Both the docstrings and dedicated documentation pages (readme file, tutorials, how-to examples, discussions, and glossary) should include cross-references to any mentioned module, class, function, method, attribute, or documentation page.
+* We use [Travis](https://travis-ci.org/) to test code.

```

Comment:
We use [Travis](https://travis-ci.org/) as a continuous integration tool, to run tests after every commit to check the model examples still work and produce the expected result.

Reply:
:+1: